graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+
+```
+type Query {
+    profile: Person!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+
+```graphql
+query getProfile {
+    profile {
+        name
+    }
+}
+```
+
+Which will result in the following JSON:
+
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White"
+        }
+    }
+}
+```
+
+#### Arrays
+A GraphQL resource can return an array of the types mentioned above.
+When a resource is returning an array, the result will be a JSON array.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get people() returns Person[] {
+    Person p1 = { name: "Walter White", age: 51 };
+    Person p2 = { name: "James Moriarty", age: 45 };
+    Person p3 = { name: "Tom Marvolo Riddle", age: 71 };
+    return [p1, p2, p3];
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: [Person!]!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    people {
+        name
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "people": [
+            {
+                "name": "Walter White"
+            },
+            {
+                "name": "James Moriarty"
+            },
+            {
+                "name": "Tom Marvolo Riddle"
+            }
+        ]
+    }
+}
+```
+
+Note that each element in the array consists only of the required `name` field.
+
+
+#### Optional Types
+A Ballerina GraphQL resource can return an optional type. When the return value is `()`, the resulting field in the JSON will be `null`.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile(int id) returns Person? {
+    if (id == 1) {
+        return { name: "Walter White", age: 51 };
+    }
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: Person
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    profile(id: 1) {
+        name
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White"
+        }
+    }
+}
+```  
+
+If the following document is used:
+```
+{
+    profile(id: 4) {
+        name
+    }
+}
+```
+
+This will be the result:
+```json
+{
+    "data": {
+        "profile": null
+    }
+}
+``` 
+
+#### Union Types
+The Ballerina GraphQL service can return a union of distinct service types. This will be mapped to a GraphQL `UNION` type.
+
+Since Ballerina supports the union types by nature, directly returning a union type is also allowed, but not recommended. The recommended way is to define a union type name separately and then use that type name, as shown in the following example. If a union type is returned directly without providing a type name, the type name will be `T1|T2|T3|...|Tn`.
```

Comment:
```suggestion
Since Ballerina supports the union types by nature, directly returning a union type is also allowed (but not recommended). The recommended way is to define a union type name separately and then use that type name as shown in the following example. If a union type is returned directly without providing a type name, the type name will be `T1|T2|T3|...|Tn`.
```